contracts/lqmining/parallel/LMPoolSelfHosted.sc

```diff
@@ -0,0 +1,288 @@
+  //
+  // ErgoTreeTemplate: d82cd601b2a5730000d602db63087201d603db6308a7d604b27203730100d605e4c6a70410d6068cc7a701d607b27202730200d608b27203730300d609b27202730400d60ab27203730500d60bb27202730600d60cb27203730700d60d8c720c01d60eb27202730800d60fb27203730900d6108c720f01d611b27202730a00d612b27203730b00d6138c720a02d614998c7209027213d615b27205730c00d616b27205730d00d6179a99a37216730ed618b27205730f00d6199d72177218d61a95919e7217721873109a721973117219d61b90721a7312d61ce4c6a70804d61d95721b731393a1721599721a7314721cd61ee4c672010804d61f93721e721cd620998c720b028c720c02d6218c720f02d622998c720e027221d623b27205731500d6248c720702d6258c720802d6268c721102d6278c721202d628e4c6a70511d629b27228731600d62ab27228731700d62b7318d62ce4c6a7061ad1edededededed93b272027319007204eded93e4c6720104107205731a9272067206731b93c27201c2a7edededed938c7207018c720801938c7209018c720a01938c720b01720d938c720e017210938c7211018c72120190b17202731c95917214731dd803d62d997e721505a2731e7e721a05d62e9c7214722dd62fb2a5731f00ededededed721d92722d7320721f937214f0722093722ef07222ededed93cbc2722f7321938602720d7214b2db6308722f7322009386027210722eb2db6308722f73230093e4c6722f060e8c720401958f72147324edededec721d92a39a9a72169c721572187223721f937220f072149272229591721a721573259c7220997e721505a273267e721a05d802d62d9972247225d62e9972267227959172227327d805d62f9a721c7328d630997215722fd6317e723005d6329999732972219c72137231d6337e9a7230732a0595917232732bd802d6347e722206d6357e723206ededededed90722f99721a732c721f907ef0722d069d9c997e9972259c9d72297233723106722b72347235907ef0722e069d9c997e9972279c9d722a7233723106722b72347235937214732d937220732e732f959372227330d807d62ff0e4c6a70705d63093722a7331d631ec909972257229722fedef723090997227722a722fd632ed94722473329472277333d633e4c672010511d634b27233733400d635b2723373350095ededec91722d733691722e7337ef72317232edededed8f721a721593723472249372357226721d721f95ed72317232d803d63695721b721595ed91721a73388f721a72159a99721572197339733ad6379972297225d63899722a7227edededec90721a733bedec7230ed9272389c722f7e723605909c7e7238067e7236067e722a06ed9272379c722f7e723605909c7e7237067e7236067e7229069372347224937235722693721e9a721c733c95edef7232ef7231d801d636b2a5733d00ed958f722d733e93c27236b2722c733f0093c27236b2722c73400092a39a9a72169c72157218722373417342
+  //
+  // ErgoTreeTemplateHash: bba2dbd2859af01376e96d616b7c8c1047f0f73ae3b5afec364ab04ffa699b37
+  //
+  // ===== Getting INPUTS data ===== //
+  // ===== Getting INPUTS data ===== //
+  val creationHeight0 = SELF.creationInfo._1
+
+  val poolNFT0        = SELF.tokens(0)
+  val poolMainReward0 = SELF.tokens(1)
+  val poolLQ0         = SELF.tokens(2)
+  val poolVLQ0        = SELF.tokens(3)
+  val poolTMP0        = SELF.tokens(4)
+  val poolOptReward0  = SELF.tokens(5)
```

Comment:
If the asset at this index is supposed to be optional, then I guess it can be absent. This code fails in that case.

Reply:
Yes, at least 1 optional reward token must be added during the initialisation with it's redeemer in the R5.